File: Golang/LetterCombinationsOfAPhoneNumber/LetterCombinationsOfAPhoneNumber_test.go
package LetterCombinationsOfAPhoneNumber

import (
	"testing"
	"reflect"
)

func TestLetterCombinations(t *testing.T) {

	if reflect.DeepEqual(letterCombinations(""), []string{}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(letterCombinations("1"), []string{}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(letterCombinations("2"), []string{"a", "b", "c"}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(letterCombinations("23"), []string{"ad", "ae", "af", "bd", "be", "bf", "cd", "ce", "cf"}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(letterCombinations("234"), []string{"adg", "adh", "adi", "aeg", "aeh", "aei",
		"afg", "afh", "afi", "bdg", "bdh", "bdi", "beg", "beh",
		"bei", "bfg", "bfh", "bfi", "cdg", "cdh", "cdi", "ceg",
		"ceh", "cei", "cfg", "cfh", "cfi"}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/LetterCombinationsOfAPhoneNumberTest.py
from unittest import TestCase

from LetterCombinationsOfAPhoneNumber import LetterCombinationsOfAPhoneNumber


class TestLetterCombinationsOfAPhoneNumber(TestCase):
    def test_letterCombinations(self):
        lc = LetterCombinationsOfAPhoneNumber()

        self.assertTrue(len(lc.letterCombinations("")) == 0)

        self.assertTrue(len(lc.letterCombinations("1")) == 0)

        self.assertEqual(set(lc.letterCombinations("2")), set(tuple(["a", "b", "c"])))

        self.assertEqual(set(lc.letterCombinations("23")),
                         set(tuple(["ad", "ae", "af", "bd", "be", "bf", "cd", "ce", "cf"])))

        self.assertEqual(set(lc.letterCombinations("234")), set(tuple(
            ["adg", "adh", "adi", "aeg", "aeh", "aei", "afg", "afh", "afi", "bdg", "bdh", "bdi", "beg", "beh", "bei",
             "bfg", "bfh", "bfi", "cdg", "cdh", "cdi", "ceg", "ceh", "cei", "cfg", "cfh", "cfi"])))
